presidio-analyzer/tests/test_analyzer_engine.py

```diff
@@ -304,10 +333,50 @@ def test_when_analyze_then_apptracer_has_value(self):
         text = "My name is Bart Simpson, and Credit card: 4095-2609-9393-4932,  my phone is 425 8829090"
         language = "en"
         entities = ["CREDIT_CARD", "PHONE_NUMBER", "PERSON"]
-        analyzer_engine_with_spacy = AnalyzerEngine(self.loaded_registry, app_tracer=self.app_tracer, enable_trace_pii=True)
-        results = analyzer_engine_with_spacy.analyze(self.unit_test_guid, text, entities, language, all_fields=False)
+        analyzer_engine_with_spacy = AnalyzerEngine(self.loaded_registry,
+                                                    app_tracer=self.app_tracer,
+                                                    enable_trace_pii=True)
+        results = analyzer_engine_with_spacy.analyze(self.unit_test_guid, text,
+                                                     entities, language,
+                                                     all_fields=False)
         assert len(results) == 3
         for result in results:
             assert result.analysis_explanation is not None
         assert self.app_tracer.get_msg_counter() == 2
         assert self.app_tracer.get_last_trace() is not None
+
+    def test_when_threshold_is_zero_all_results_pass(self):
+        text = " Credit card: 4095-2609-9393-4932,  my phone is 425 8829090"
+        language = "en"
+        entities = ["CREDIT_CARD", "PHONE_NUMBER"]
+
+        # This analyzer engine is different from the global one, as this one
+        # also loads SpaCy so it can detect the phone number entity
+
+        analyzer_engine = AnalyzerEngine(
+            registry=self.loaded_registry, nlp_engine=MockNlpEngine())
+        results = analyzer_engine.analyze(self.unit_test_guid, text,
+                                          entities, language,
+                                          all_fields=False,
+                                          score_threshold=0)
+
+        assert len(results) == 2
+
+    def test_when_threshold_is_half_only_credit_card_passes(self):
+        text = " Credit card: 4095-2609-9393-4932,  my phone is 425 8829090"
+        language = "en"
+        entities = ["CREDIT_CARD", "PHONE_NUMBER"]
+
+        # This analyzer engine is different from the global one, as this one
+        # also loads SpaCy so it can detect the phone number entity
+
+        analyzer_engine = AnalyzerEngine(
+            registry=self.loaded_registry, nlp_engine=MockNlpEngine())
+        results = analyzer_engine.analyze(self.unit_test_guid, text,
+                                          entities, language,
+                                          all_fields=False,
+                                          score_threshold=0.6)
```

Comment:
0.6 -> 0.5 and change the threshold statement to >=
I think it makes more sense, also the name of this test says: "...is_half..."

Reply:
I changed the test name. It has to be more than 0.5 because of the internal confidence of the pattern